src/App.tsx

```diff
@@ -1,26 +1,169 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { wait } from './utils/fetchClient';
+import { UserWarning } from './UserWarning';
+import { Footer } from './components/Footer';
+import { creatTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { getFilterTodos } from './utils/getFilterTodos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
+import { Errors } from './types/Errors';
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTitle, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [deletedTodoId, setDeletedTodoId] = useState<number>(0);
+
+  const focusInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(Errors.Load);
+
+        wait(3000).then(() => setErrorMessage(''));
+      });
+  }, []);
+
+  useEffect(() => {
+    focusInput.current?.focus();
+  }, [errorMessage, todos]);
+
+  const filteredTodos = useMemo(() => {
+    return getFilterTodos(todos, status);
+  }, [todos, status]);
+
+  const handleClearError = () => setErrorMessage('');
+
+  const filterUncompletedTodos = (): Todo[] => {
+    return todos.filter(item => !item.completed);
+  };
```

Comment:
You don't need this here, you can do it inside component where you use this value
And it should be variable, not function